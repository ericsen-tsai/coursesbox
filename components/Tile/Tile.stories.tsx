import { expect } from '@storybook/jest'
import { screen } from '@storybook/testing-library'
import { ComponentStoryObj, ComponentMeta } from '@storybook/react'

import Tile from './Tile'

export default {
  title: 'Content/Tile',
  component: Tile,
} as ComponentMeta<typeof Tile>

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  play: async () => {
    await expect(screen.getByRole('heading')).toBeInTheDocument()
  },
  args: {
    header: 'Lorem ipsum dolor sit',
    children: `
        Lorem 
        ipsum dolor sit, amet consectetur 
        adipisicing elit. Exercitationem amet distinctio facere iusto aliquam quisquam magnam, ullam labore corporis necessitatibus maiores, minus reiciendis doloremque quibusdam! Vero possimus accusamus explicabo tempora?
    `,
  },
}
