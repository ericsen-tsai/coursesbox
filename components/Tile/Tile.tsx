import React, { ReactNode } from 'react'
import styled from '@emotion/styled'

import { boxShadow, borderRadius } from '@/components/styles'

type Props = {
  header: string
  children: ReactNode
}

const Section = styled.section`
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)}
  ${borderRadius}
  color: ${({ theme }) => theme.font.regular};
  padding: 1vmin 4vmin 4vmin;
`

function Tile({ header, children }: Props) {
  return (
    <Section>
      <h2>{header}</h2>
      {children}
    </Section>
  )
}

export default Tile
