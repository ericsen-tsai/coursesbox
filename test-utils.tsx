import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
