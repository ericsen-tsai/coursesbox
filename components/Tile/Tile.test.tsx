import { render } from '@/test-utils'

import Tile from '.'

describe('Tile test cases', () => {
  it('Tile render check', () => {
    const { asFragment } = render(<Tile header="hello">World!!!</Tile>)

    expect(asFragment()).toMatchSnapshot()
  })
})
