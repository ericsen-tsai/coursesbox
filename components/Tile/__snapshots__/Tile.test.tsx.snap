// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases Tile render check 1`] = `
<DocumentFragment>
  <section
    class="css-18zumgu"
  >
    <h2>
      hello
    </h2>
    World!!!
  </section>
</DocumentFragment>
`;
